test/access-control.integration.js

```diff
@@ -226,7 +238,101 @@ describe('access control - integration', function() {
+      return '/api/accounts/' + this.accountWithReplaceOnPUTtrue.id;
+    }
+  });
+
+  describe('/accounts with replaceOnPUT false', function() {
+    var count = 0;
+    before(function() {
+      var roleModel = loopback.getModelByType(loopback.Role);
+      roleModel.registerResolver('$dummy', function(role, context, callback) {
+        process.nextTick(function() {
+          if (context.remotingContext) {
+            count++;
+          }
+          callback && callback(null, false); // Always true
+        });
+      });
+    });
+
+    lt.beforeEach.givenModel('accountWithReplaceOnPUTfalse');
+
+    lt.it.shouldBeDeniedWhenCalledAnonymously('GET', '/api/accounts2');
+    lt.it.shouldBeDeniedWhenCalledUnauthenticated('GET', '/api/accounts2');
+    lt.it.shouldBeDeniedWhenCalledByUser(CURRENT_USER, 'GET', '/api/accounts2');
+
+    lt.it.shouldBeDeniedWhenCalledAnonymously('GET', urlForAccount);
+    lt.it.shouldBeDeniedWhenCalledUnauthenticated('GET', urlForAccount);
+    lt.it.shouldBeDeniedWhenCalledByUser(CURRENT_USER, 'GET', urlForAccount);
+
+    lt.it.shouldBeDeniedWhenCalledAnonymously('POST', '/api/accounts2');
+    lt.it.shouldBeDeniedWhenCalledUnauthenticated('POST', '/api/accounts2');
+    lt.it.shouldBeDeniedWhenCalledByUser(CURRENT_USER, 'POST', '/api/accounts2');
+
+    lt.it.shouldBeDeniedWhenCalledAnonymously('POST', urlForReplaceAccountPOST);
+    lt.it.shouldBeDeniedWhenCalledUnauthenticated('POST', urlForReplaceAccountPOST);
+    lt.it.shouldBeDeniedWhenCalledByUser(CURRENT_USER, 'POST', urlForReplaceAccountPOST);
+
+    lt.it.shouldBeDeniedWhenCalledAnonymously('PATCH', urlForAccount);
+    lt.it.shouldBeDeniedWhenCalledUnauthenticated('PATCH', urlForAccount);
+    lt.it.shouldBeDeniedWhenCalledByUser(CURRENT_USER, 'PATCH', urlForAccount);
+
+    lt.describe.whenLoggedInAsUser(CURRENT_USER, function() {
+      beforeEach(function(done) {
+        var self = this;
+        // Create an account under the given user
+        app.models.accountWithReplaceOnPUTfalse.create({
+          userId: self.user.id,
+          balance: 100,
+        }, function(err, act) {
+          self.url = '/api/accounts2/' + act.id;
+
+          done();
+        });
+      });
+
+      lt.describe.whenCalledRemotely('PATCH', '/api/accounts2/:id', function() {
+        lt.it.shouldBeAllowed();
+      });
+
+      lt.describe.whenCalledRemotely('PUT', '/api/accounts2/:id', function() {
+        lt.it.shouldBeAllowed();
+      });
+      lt.describe.whenCalledRemotely('GET', '/api/accounts2/:id', function() {
+        lt.it.shouldBeAllowed();
+      });
+      lt.describe.whenCalledRemotely('DELETE', '/api/accounts2/:id', function() {
+        lt.it.shouldBeDenied();
+      });
+
+      describe('replace on POST verb', function() {
+        beforeEach(function(done) {
+          var self = this;
+          // Create an account under the given user
+          app.models.accountWithReplaceOnPUTfalse.create({
```

Comment:
Good question....That was tricky; I copy & paste the `create` method from above L285 and `create` method here for `replace on POST verb here` (because after applying feedbacks the lines keep changing);

create method from above L285:
```
        app.models.accountWithReplaceOnPUTfalse.create({
          userId: self.user.id,
          balance: 100,
        }, function(err, act) {
          self.url = '/api/accounts2/' + act.id;

          done();
        });
```

create method here for `replace on POST verb here`:
```
          app.models.accountWithReplaceOnPUTfalse.create({
            userId: self.user.id,
            balance: 100,
          }, function(err, act) {
            self.url = '/api/accounts2/' + act.id + '/replace';
            done();
          });
```

As you see after creating an account we store url on `this.url`and that is used in [whenCalledRemotely](https://github.com/strongloop/loopback/blob/70cec0755acc2b670987675c3bc63ecbfe9b473f/test/helpers/loopback-testing-helper.js#L157) in `loopback-testing-helper.js`; please note that the url for `replace` has `/replace` at the end, so thats why I created another account for testing `replace` method, to have the right url when we call `whenCalledRemotely` in `loopback-testing-helper.js`